### PR TITLE
CME-53: Java 21

### PR DESCRIPTION
**Pull request by AntonyLeons**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CME-53

### Change description ###

also bring appinsights in line to to help investigate [CME-195](https://tools.hmcts.net/jira/browse/CME-195)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: infrastructure/aat.tfvars [hunk 1]
```diff
@@ -0,0 +1,2 @@
+aks_subscription_id = "96c274ce-846d-4e48-89a7-d528432298a7"
```
Comment (kiran-yenigala-hmcts) [inline infrastructure/aat.tfvars:1]: what is causing this change, we are just upgrading Java 21 in this PR?

**Review by kiran-yenigala-hmcts — APPROVED** [review]
LGTM

File: infrastructure/variables.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
+variable "product" {
+  default = "hmc"
+}
+
+variable "component" {
+  default = "hmi-inbound-adapter"
```
Comment (RebeccaBaker) [inline infrastructure/variables.tf:6]: should be outbound-adapter

**Review by lang-ben — APPROVED** [review]
lgtm

**Pull request merged**